javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -253,37 +315,198 @@ object Character {
 
-  def isUnicodeIdentifierStart(c: scala.Char): scala.Boolean =
-    reUnicodeIdentStart.test(c.toString)
-
-  def isUnicodeIdentifierPart(c: scala.Char): scala.Boolean =
-    isUnicodeIdentifierStart(c) || isIdentifierIgnorable(c) ||
-    reUnicodeIdentPartExcl.test(c.toString)
-
-  def isIdentifierIgnorable(c: scala.Char): scala.Boolean =
-    reIdentIgnorable.test(c.toString)
+  def isLowerCase(c: scala.Char): scala.Boolean =
+    isLowerCase(c.toInt)
 
-  //def isMirrored(c: scala.Char): scala.Boolean
+  def isLowerCase(c: Int): scala.Boolean = {
+    if (c < 256)
+      c == '\u00AA' || c == '\u00BA' || getTypeLT256(c) == LOWERCASE_LETTER
+    else
+      isLowerCaseGE256(c)
+  }
 
-  def isLowerCase(c: scala.Char): scala.Boolean =
-    reIsLower.test(c.toString)
+  private[this] def isLowerCaseGE256(c: Int): scala.Boolean = {
+    ('\u02B0' <= c && c <= '\u02B8') || ('\u02C0' <= c && c <= '\u02C1') ||
+    ('\u02E0' <= c && c <= '\u02E4') || c == '\u0345' || c == '\u037A' ||
+    ('\u1D2C' <= c && c <= '\u1D6A') || c == '\u1D78' ||
+    ('\u1D9B' <= c && c <= '\u1DBF') || c == '\u2071' || c == '\u207F' ||
+    ('\u2090' <= c && c <= '\u209C') || ('\u2170' <= c && c <= '\u217F') ||
+    ('\u24D0' <= c && c <= '\u24E9') || ('\u2C7C' <= c && c <= '\u2C7D') ||
+    c == '\uA770' || ('\uA7F8' <= c && c <= '\uA7F9') ||
+    getTypeGE256(c) == LOWERCASE_LETTER
+  }
 
   def isUpperCase(c: scala.Char): scala.Boolean =
-    reIsUpper.test(c.toString)
+    isUpperCase(c.toInt)
+
+  def isUpperCase(c: Int): scala.Boolean = {
+    ('\u2160' <= c && c <= '\u216F') || ('\u24B6' <= c && c <= '\u24CF') ||
+    getType(c) == UPPERCASE_LETTER
+  }
 
   @inline def isValidCodePoint(codePoint: Int): scala.Boolean =
     codePoint >= MIN_CODE_POINT && codePoint <= MAX_CODE_POINT
 
+  @inline def isBmpCodePoint(codePoint: Int): scala.Boolean =
+    codePoint >= MIN_VALUE && codePoint <= MAX_VALUE
+
   @inline def isSupplementaryCodePoint(codePoint: Int): scala.Boolean =
     codePoint >= MIN_SUPPLEMENTARY_CODE_POINT && codePoint <= MAX_CODE_POINT
 
-  //def isTitleCase(c: scala.Char): scala.Boolean
-  //def isJavaIdentifierPart(c: scala.Char): scala.Boolean
+  def isTitleCase(c: scala.Char): scala.Boolean =
+    isTitleCase(c.toInt)
+
+  def isTitleCase(cp: Int): scala.Boolean =
+    if (cp < 256) false
+    else isTitleCaseImpl(getTypeGE256(cp))
+
+  @inline private[this] def isTitleCaseImpl(tpe: Int): scala.Boolean =
+    tpe == TITLECASE_LETTER
+
+  def isDigit(c: scala.Char): scala.Boolean =
+    isDigit(c.toInt)
+
+  def isDigit(cp: Int): scala.Boolean =
+    if (cp < 256) '0' <= cp && cp <= '9'
+    else isDigitImpl(getTypeGE256(cp))
+
+  @inline private[this] def isDigitImpl(tpe: Int): scala.Boolean =
+    tpe == DECIMAL_DIGIT_NUMBER
+
+  def isDefined(c: scala.Char): scala.Boolean =
+    isDefined(c.toInt)
+
+  def isDefined(c: scala.Int): scala.Boolean =
+    if (c < 0) false
+    else if (c < 888) true
+    else getTypeGE256(c) != UNASSIGNED
```

Comment:
To be completely precise, this method needs `{}`. Since it has 3 lines, it's not a two-liner if/else.